airgun/widgets.py

```diff
@@ -2100,7 +2100,20 @@ class AutoCompleteTextInput(TextInput):
     force the auto-completion list to be hidden.
     """
 
+    clear_button = Text(
+        locator="//span[contains(@class,'autocomplete-clear-button') or "
+                "contains(@class,'fa-close')]"
+    )
+
+    def clear(self):
+        """Clears search field value and re-trigger search to remove all
+        filters.
+        """
+        if self.clear_button.is_displayed:
+            self.clear_button.click()
+
     def fill(self, value):
+        self.clear()
         changes = super().fill(value)
```

Comment:
If we were to follow widgetastic usage guidelines to the letter, calling `self.clear()` might change value, so `changes` here should be combined with `self.clear()` output in some way.

I mean situation like this:
1. widget has some content
2. I call `widget.fill('')` with intent of clearing this content
3. `fill()` calls `clear()`, clearing the content
4. `super().fill('')` is called. Since content is already cleared, it doesn't change, so returns False
5. my initial `widget.fill('')` returns False, signifying that widget content was not changed - which is not true at all

I'm personally not very concerned about this, but something to keep in mind.